tools/site_admin/project_jump.php

```diff
@@ -0,0 +1,137 @@
+include_once($relPath.'Project.inc');
+include_once($relPath.'project_events.inc');
+
+require_login();
+
+if (!user_is_a_sitemanager()) {
+    die("You are not authorized to invoke this script.");
+}
+
+$action = get_enumerated_param($_POST, 'action', 'showform', ['showform', 'check', 'dojump']);
+$projectid = get_projectID_param($_POST, 'projectid', true);
+$valid_new_states = [];
+foreach ($Round_for_round_id_ as $id => $round) {
+    $valid_new_states[] = $round->project_unavailable_state;
+}
+$new_state = get_enumerated_param($_POST, 'new_state', null, $valid_new_states, true);
+
+
+$title = _("Project Jump");
+output_header($title);
+echo "<h1>$title</h1>";
+
+echo "<p>" . _("Jump a project to a specific state.") . "</p>\n";
+
+switch ($action) {
+    case 'showform':
+        display_form("showform", $projectid, $new_state);
+        break;
+
+    case 'check':
+        do_stuff($projectid, $new_state, true);
+        display_form("check", $projectid, $new_state);
+        break;
+
+    case 'dojump':
+        do_stuff($projectid, $new_state, false);
+        echo "\n\n<a href='$code_url/project.php?id={$projectid}'>" . _("Project Home") . "</a>\n";
+        break;
+}
+
+
+function display_form($action, $projectid, $new_state)
+{
+    global $Round_for_round_id_;
+
+    echo "<form method='post'>\n";
+    echo "<table>\n";
+    if ($action == "showform") {
+        echo "<tr>\n";
+        echo "<th>" . _("Project") . ":</th>\n";
+        echo "<td><input type='text' name='projectid' size='28' required></td>\n";
+        echo "</tr>";
+
+        echo "<tr>";
+        echo "<th>" . _("State") . ":</th>\n";
+        echo "<td><select name='new_state'>";
+        foreach ($Round_for_round_id_ as $id => $round) {
+            echo "<option value='{$round->project_unavailable_state}'>{$round->project_unavailable_state}</option>";
+        }
+        echo "</select></td>\n";
+        echo "</tr>";
+
+        echo "<tr>";
+        echo "<td></td>";
+        echo "<td>";
+        echo "<input type='submit' name='submit_button' value='" . attr_safe(_("Check")) . "'>\n";
+        echo "<input type='hidden' name='action' value='check'>\n";
+        echo "</td>\n";
+        echo "</tr>";
+    } elseif ($action == "check") {
+        echo "<tr>";
+        echo "<td>";
+        echo "<input type='hidden' name='projectid' value='" . attr_safe($projectid) . "'>";
+        echo "<input type='hidden' name='new_state' value='" . attr_safe($new_state) . "'>";
+        echo "<input type='hidden' name='action' value='dojump'>\n";
+        echo "<input type='submit' name='submit_button' value='" . attr_safe(_("Do it")) . "'>";
+        echo "</td>";
+        echo "</tr>";
+    }
+    echo "</table>\n";
+    echo "</form>";
+}
+
+function do_stuff($projectid, $new_state, $just_checking)
+{
+    global $pguser;
+    $project = new Project($projectid);
+    echo "<pre>";
+    echo "    projectid : $projectid\n";
+    echo "    title     : $project->nameofwork\n";
```

Comment:
should we encode the name of work in case there is a </pre> in it? I know, I know...

Reply:
We should, but all of the `site_admin/` scripts need TLC to not be using `<pre>` tags at all. So in this case I'd rather have the consistency. The SAs are use to them at this point :/

Reply:
Speaking as an SA, I assume that the reason the `<pre>` wrappers were used in the first place was to make the text stand out, and nobody thought things through enough to use CSS when setting the scripts up. Most of them pre-date the update of the site-wide CSS.

I was assuming that in this case, @chrismiceli's comment was aimed at avoiding a situation like this (I added a `</pre>` to the project title):
<img width="323" alt="image" src="https://user-images.githubusercontent.com/36800239/146724867-71ab0211-cc9f-49a6-af9e-560bda829734.png">

where the PM included a `</pre>` in the nameofwork field. Which kind of breaks the display of the title. The title is actually "Book clone for deletion".

Reply:
That was my point, but I think I agree with @cpeel and we can do this all at once in another MR if we want since project creation is limited to ideally non malicious actors who won't embed XSS type attacks.

Reply:
👍  -- except for those of us doing the testing and going out of our way to break things. 😁